diff --git a/config.def.h b/config.def.h
index 82b1b09..329a1d8 100644
--- a/config.def.h
+++ b/config.def.h
@@ -6,7 +6,7 @@
  * font: see http://freedesktop.org/software/fontconfig/fontconfig-user.html
  */
 static char *font = "Liberation Mono:pixelsize=12:antialias=true:autohint=true";
-static int borderpx = 2;
+static float borderchars = 2;
 
 /*
  * What program is execed by st depends of these precedence rules:
diff --git a/st.h b/st.h
index dac64d8..629d081 100644
--- a/st.h
+++ b/st.h
@@ -120,4 +120,5 @@ extern int allowaltscreen;
 extern char *termname;
 extern unsigned int tabspaces;
 extern unsigned int defaultfg;
+int borderpx;
 extern unsigned int defaultbg;
diff --git a/x.c b/x.c
index c343ba2..f6b63d6 100644
--- a/x.c
+++ b/x.c
@@ -953,6 +953,8 @@ xloadfonts(char *fontstr, double fontsize)
 	win.cw = ceilf(dc.font.width * cwscale);
 	win.ch = ceilf(dc.font.height * chscale);
 
+	borderpx = ceilf(borderchars * win.cw);
+
 	FcPatternDel(pattern, FC_SLANT);
 	FcPatternAddInteger(pattern, FC_SLANT, FC_SLANT_ITALIC);
 	if (xloadfont(&dc.ifont, pattern))
